### PR TITLE
fix: don't return ReadException for failed json parsing

### PR DESCRIPTION
**Pull request by hampuslavin**
## Description
- Moves json parsing to the `try` clause that actually throws a `DeserializationException`

### CODE REVIEW

File: test/local_storage_manager/local_storage_manager_test.dart [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      () => LocalStorageManager.tryFetchAndDeserializeJsonFile<
+          Map<String, dynamic>>(
+        fileName: fileName,
+        localStoragePath: localStoragePath,
+        fromJson: (json) => json,
+      ),
+      throwsA(isA<DeserializationException>()),
+    );
+  });
+}
```
Comment (hampuslavin) [inline test/local_storage_manager/local_storage_manager_test.dart:135]: It wasn't straight forward to simulate a failed read in the test 🧐 Any idea?

**Review by SandPod — COMMENTED** [review]
LGTM! Given the addition of the read test we discussed AFK(IRL).

File: test/local_storage_manager/local_storage_manager_test.dart [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      localStoragePath: localStoragePath,
+    );
+
+    var file = File(p.join(localStoragePath, fileName));
+    expect(file.existsSync(), isTrue);
+    expect(file.readAsStringSync(), '{\n  "key": "value"\n}');
+  });
+
+  test(
+      'Given a json serialization error '
```
[the remark this answers is not in the document]
Comment (hampuslavin) [reply test/local_storage_manager/local_storage_manager_test.dart]: e472ddc

File: test/local_storage_manager/local_storage_manager_test.dart [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    var file = File(p.join(localStoragePath, fileName));
+    file.createSync();
+
+    await LocalStorageManager.removeFile(
+      fileName: fileName,
+      localStoragePath: localStoragePath,
+    );
+
+    expect(file.existsSync(), isFalse);
+  });
```
[the remark this answers is not in the document]
Comment (hampuslavin) [reply test/local_storage_manager/local_storage_manager_test.dart:29]: Couldn't find any test fil for this class but maybe I'm missing it? 🤷 🧐

File: test/local_storage_manager/local_storage_manager_test.dart [hunk 4]
```diff
@@ -0,0 +1,135 @@
+import 'dart:io';
+import 'package:test/test.dart';
+import 'package:path/path.dart' as p;
+import 'package:cli_tools/src/local_storage_manager/local_storage_manager_exceptions.dart';
+import 'package:cli_tools/src/local_storage_manager/local_storage_manager.dart';
+
+void main() {
+  var tempDir = Directory.systemTemp.createTempSync();
```
[the remark this answers is not in the document]
Comment (hampuslavin) [reply test/local_storage_manager/local_storage_manager_test.dart:8]: It only needs to happen once so I don't think so?

File: test/local_storage_manager/local_storage_manager_test.dart (hunk 1 above)
Comment (hampuslavin) [reply test/local_storage_manager/local_storage_manager_test.dart:154]: solved: e472ddc

**Review by SandPod — APPROVED** [review]
LGTM! 👍

File: test/local_storage_manager/local_storage_manager_test.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (SandPod) [reply test/local_storage_manager/local_storage_manager_test.dart:29]: Might have missed adding it when creating this project 🙈

File: test/local_storage_manager/local_storage_manager_test.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (SandPod) [reply test/local_storage_manager/local_storage_manager_test.dart:8]: Cool! No then it is probably enough.

**Pull request merged**